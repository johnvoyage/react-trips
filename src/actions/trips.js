import uuid from 'uuid';
import database from '../firebase/firebase';

export const startAddTrip = (tripData = {}) => {
  return (dispatch) => {
    const {
      country = ''
    } = tripData;
    const trip = {
      country
    };

    return database.ref('trips').push(trip).then((ref) => {
      dispatch(addTrip({
        id: ref.key,
        ...trip
      }));
    });
  };
};

export const addTrip = (trip) => ({
  type: 'ADD_TRIP',
  trip
});

export const removeTrip = ({ id } = {}) => ({
  type: 'REMOVE_TRIP',
  id
});

export const editTrip = (id, updates) => ({
  type: 'EDIT_TRIP',
  id,
  updates
});
