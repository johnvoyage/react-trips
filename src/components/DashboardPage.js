import React from 'react';

const DashboardPage = () => (
  <p>Dashboard page</p>
);

export default DashboardPage;