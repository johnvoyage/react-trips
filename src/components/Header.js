import React from 'react';
import { NavLink } from 'react-router-dom';


const Header = () => (
  <header>
    <h1>header</h1>
    <NavLink to="/" activeClassName='is-active' exact={true}>Home</NavLink>
    <NavLink to="/trips" activeClassName='is-active' exact={true}>Trips</NavLink>
    <NavLink to="/trips/new" activeClassName='is-active' exact={true}>New Trip</NavLink>

    <NavLink to="/profile" activeClassName='is-active'>Profile</NavLink>
  </header>
);
export default Header;