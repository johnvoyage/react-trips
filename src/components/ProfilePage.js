import React from 'react';

const ProfilePage = (props) => {
  console.log(props);
  return (
    <p>profile page for user: {props.match.params.id}</p>
  );
};

export default ProfilePage;