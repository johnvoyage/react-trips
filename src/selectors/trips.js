export const getVisibleTrips = (trips, { text, sortBy, startDate, endDate }) => {
  return trips.filter((trip) => {
    const startDateMatch = typeof startDate !== 'number' || trip.createdAt >= startDate;
    const endDateMatch = typeof endDate !== 'number' || trip.createdAt <= endDate;
    const textMatch = trip.country.toLowerCase().includes(text.toLowerCase());

    return startDateMatch && endDateMatch && textMatch;
  }).sort((a, b) => {
    if (sortBy === 'date') {
      return a.createdAt < b.createdAt ? 1 : -1;
    } else {
      return 1;
    }
    //else if (sortBy === 'amount') {
    //   return a.amount < b.amount ? 1 : -1;
    // }
  });
};

