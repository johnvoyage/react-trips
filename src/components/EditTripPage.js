import React from 'react';
import { connect } from 'react-redux';

import TripForm from './TripForm';
import { editTrip, removeTrip } from '../actions/trips';

const EditTripPage = (props) => (
  <div>
    <TripForm
      trip={props.trip} 
      onSubmit={(trip) => {
        props.dispatch(editTrip(props.trip.id, trip));
        props.history.push('/trips');
      }}
    />
    <button onClick={() => {
      props.dispatch(removeTrip({ id: props.trip.id }));
      props.history.push('/trips');
    }}>Remove</button>
  </div>
);

const mapStateToProps = (state, props) => {
  return {
    trip: state.trips.find((trip) => trip.id === props.match.params.id)
  };
};

export default connect(mapStateToProps)(EditTripPage);