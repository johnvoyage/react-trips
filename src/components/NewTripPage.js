import React from 'react';
import { connect } from 'react-redux';

import TripForm from './TripForm';
import { addTrip } from '../actions/trips';

export class NewTripPage extends React.Component {
  onSubmit = (trip) => {
    // props.dispatch(addTrip(trip));
    this.props.addTrip(trip)
    this.props.history.push('/trips');
  };

  render() {
    return (
      <div> 
        <h1>New Trip</h1>
        <TripForm 
          onSubmit={this.onSubmit}
        />
      </div>
    )
  }
};

const mapDispatchToProps = (dispatch) => ({
    addTrip: (trip) => dispatch(addTrip(trip))
});

export default connect(undefined, mapDispatchToProps)(NewTripPage);