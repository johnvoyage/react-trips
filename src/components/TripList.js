import React from 'react';
import { connect } from 'react-redux';

import TripListItem from './TripListItem';
import { getVisibleTrips } from '../selectors/trips';

export const TripList = (props) => (
  <div>
    {
      props.trips.length === 0 ? (
        <p>No trips</p>
      ) : (
        props.trips.map((trip) => <TripListItem key={trip.id} {...trip} />)
      )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    trips: getVisibleTrips(state.trips, state.filters)
  };
};

export default connect(mapStateToProps)(TripList);
