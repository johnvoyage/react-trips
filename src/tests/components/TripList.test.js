import React from 'react';
import { shallow } from 'enzyme';

import trips from '../fixtures/trips';
import { TripList } from '../../components/TripList';

test('should render TripList with trips', () => {
  const wrapper = shallow(<TripList trips={trips}/>)
  expect(wrapper).toMatchSnapshot();
});

test('should render TripList with no trips', () => {
  const wrapper = shallow(<TripList trips={[]}/>)
  expect(wrapper).toMatchSnapshot();
});