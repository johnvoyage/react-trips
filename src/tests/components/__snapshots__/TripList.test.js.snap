// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TripList with no trips 1`] = `
<div>
  <p>
    No trips
  </p>
</div>
`;

exports[`should render TripList with trips 1`] = `
<div>
  <TripListItem
    country="Germany"
    id="123abc"
    key="123abc"
  />
  <TripListItem
    country="USA"
    id="sfasg"
    key="sfasg"
  />
  <TripListItem
    country="Uruguay"
    id="gdsash"
    key="gdsash"
  />
</div>
`;
